lua/nvim_lsp/jsonls.lua

```diff
@@ -0,0 +1,54 @@
+local skeleton = require 'nvim_lsp/skeleton'
+local util = require 'nvim_lsp/util'
+local lsp = vim.lsp
+
+local server_name = "jsonls"
+local bin_name = "vscode-json-languageserver"
+
+local installer = util.npm_installer {
+  server_name = server_name;
+  packages = {bin_name};
+  binaries = {bin_name};
+}
+
+skeleton[server_name] = {
+  default_config = util.utf8_config {
+    cmd = {bin_name, "--stdio"};
+    filetypes = {"json"};
+    root_dir = util.root_pattern(".git", vim.fn.getcwd());
+    log_level = lsp.protocol.MessageType.Warning;
+    settings = {};
+  };
+  on_new_config = function(new_config)
+    local install_info = installer.info()
+    if install_info.is_installed then
+      if type(new_config.cmd) == 'table' then
+        -- Try to preserve any additional args from upstream changes.
+        new_config.cmd[1] = install_info.binaries[bin_name]
+      else
+        new_config.cmd = {install_info.binaries[bin_name]}
+      end
+    end
+  end;
+  docs = {
+    -- this language server config is in VSCode built-in package.json
+    package_json = "https://raw.githubusercontent.com/microsoft/vscode/master/extensions/json-language-features/package.json";
```

Comment:
Use `vscode` key?

Reply:
I wrote it in the code comment, but this extension is built-in to vscode, so it is not distributed as a package.